### PR TITLE
ROC-5160 Split Full Name into title, first, last name for Individual and Sole Traders (their details)

### PR DESCRIPTION
**Pull request by simad**
This PR adds support for split title(optional) / first name / last name (rather than full name) for their details, individuals and sole traders.

Right now, frontend sends full name instead of the split version, so https://github.com/hmcts/cmc-claim-store/pull/829/commits/fd9c63576bb361e96bff5d5bfbeb779c6093a001 adds some backward compatibility fixes which will be removed once the frontend pr (https://github.com/hmcts/cmc-citizen-frontend/pull/1539) for this story is merged to master

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5160
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class SplitNameConstraintValidator implements ConstraintValidator<SplitName, TheirDetails> {
+
+    @Override
+    public boolean isValid(TheirDetails theirDetails, ConstraintValidatorContext context) {
+        if (theirDetails instanceof SplitNamedParty) {
+            if (StringUtils.isBlank(theirDetails.getName()) &&
+                (StringUtils.isAnyBlank(((SplitNamedParty) theirDetails).getFirstName(), ((SplitNamedParty) theirDetails).getLastName()))) {
+                return false;
+            }
+            return true;
```
Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java]: Could directly return the negation of the `if` condition

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.regex.Pattern;
+// todo ROC-5160 delete this class once frontend is merged
+
+public class SplitNameConstraintValidator implements ConstraintValidator<SplitName, TheirDetails> {
+
+    @Override
+    public boolean isValid(TheirDetails theirDetails, ConstraintValidatorContext context) {
+        if (theirDetails instanceof SplitNamedParty) {
+            if (StringUtils.isBlank(theirDetails.getName()) &&
+                (StringUtils.isAnyBlank(((SplitNamedParty) theirDetails).getFirstName(), ((SplitNamedParty) theirDetails).getLastName()))) {
```
[the remark this answers is not in the document]
Comment (simad) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java]: 👍

[the remark this answers is not in the document]
Comment (simad) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java]: the whole class will go away once frontend is merged, so I didn't really focus on making it nice :D

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java (hunk 1 above)
Comment (simad) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java]: 👍 same as above - will get rid of this whole class :D

**Pull request merged**